test/e2e/gc_e2e_test.go

```diff
@@ -34,159 +33,268 @@ var _ = Describe("Garbage collector", func() {
+							return err
+						})
+						Expect(err).NotTo(HaveOccurred())
+
+						By("should not have deleted the dependent since ownerB CSV is still present", func() {
+							_, err := kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Get(context.TODO(), dependent.GetName(), metav1.GetOptions{})
+							Expect(err).NotTo(HaveOccurred(), "dependent deleted after one of the owner was deleted")
+							ctx.Ctx().Logf("dependent still exists after one owner was deleted")
+
+						})
+					})
+
+					It("deleting ownerB CSV", func() {
+
+						// delete ownerB in the foreground (to ensure any "blocking" dependents are deleted before ownerB)
+						err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(context.TODO(), fetchedB.GetName(), options)
+						Expect(err).NotTo(HaveOccurred())
+
+						// wait for deletion of ownerB
+						err = waitForDelete(func() error {
+							_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), ownerB.GetName(), metav1.GetOptions{})
+							return err
+						})
+						Expect(err).NotTo(HaveOccurred())
+
+						By("should not have deleted the dependent since ownerA CSV is still present", func() {
+
+							_, err := kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Get(context.TODO(), dependent.GetName(), metav1.GetOptions{})
+							Expect(err).NotTo(HaveOccurred(), "dependent deleted after one of the owner was deleted")
+
+							ctx.Ctx().Logf("dependent still exists after one owner was deleted")
+						})
+					})
+				})
+
+				When("removing both the owners", func() {
+
+					It("deleting both ownerA and ownerB CSVs", func() {
+
+						// delete ownerA in the foreground (to ensure any "blocking" dependents are deleted before ownerA)
+						err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(context.TODO(), fetchedA.GetName(), options)
+						Expect(err).NotTo(HaveOccurred())
+
+						// wait for deletion of ownerA
+						err = waitForDelete(func() error {
+							_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), ownerA.GetName(), metav1.GetOptions{})
+							return err
+						})
+						Expect(err).NotTo(HaveOccurred())
+
+						// delete ownerB in the foreground (to ensure any "blocking" dependents are deleted before ownerB)
+						err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(context.TODO(), fetchedB.GetName(), options)
+						Expect(err).NotTo(HaveOccurred())
+
+						// wait for deletion of ownerB
+						err = waitForDelete(func() error {
+							_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), ownerB.GetName(), metav1.GetOptions{})
+							return err
+						})
+						Expect(err).NotTo(HaveOccurred())
+
+						By("should have deleted the dependent since both the owners were deleted", func() {
```

Comment:
This right here is the spec: "It should delete the dependent".

I'd also suggest that we don't pass the optional closure to `ginkgo.By`. It gives the impression that a new container or spec node is being created, but it doesn't actually work that way.